common/persistence/sql/sqlExecutionManager.go

```diff
@@ -1417,25 +1563,22 @@ func createCurrentExecution(tx *sqlx.Tx, request *p.CreateWorkflowExecutionReque
 		State:           p.WorkflowStateRunning,
 		CloseStatus:     p.WorkflowCloseStatusNone,
 	}
-	if request.ReplicationState != nil {
-		arg.StartVersion = request.ReplicationState.StartVersion
-		arg.LastWriteVersion = request.ReplicationState.LastWriteVersion
-	} else {
-		arg.StartVersion = common.EmptyVersion
-		arg.LastWriteVersion = common.EmptyVersion
+	replicationState := request.ReplicationState
+	if replicationState != nil {
+		arg.StartVersion = &replicationState.StartVersion
+		arg.LastWriteVersion = &replicationState.LastWriteVersion
 	}
 	if request.ParentExecution != nil {
 		arg.State = p.WorkflowStateCreated
 	}
 
 	switch request.CreateWorkflowMode {
 	case p.CreateWorkflowModeContinueAsNew:
-		if err := continueAsNew(tx,
+		if err := conditionalUpdateCurrentExecution(tx,
```

Comment:
I don't think the method "conditionalUpdateCurrentExecution" actually does a conditional update. Name just adds confusion.